Apply clippy::uninlined_format_args fixes

nyurik opened the pull request:
This is the result of running `clippy::uninlined_format_args` lint. Currently the lint is in `pedantic`, but there are plans/hopes to move it to `style`.

How can I add this specific lint to the current build/CI tools, or is this not possible?  It would be ideal not to make the lint as the default style and then end up with hundreds of changes in an unrelated PR due to CI requirements.

Locally, I ran this to generate the changes.  Not ideal because it has produced a number of compilation errors.

```bash
rustup run nightly cargo clippy --fix --no-deps --allow-dirty -- -A clippy::all -W clippy::uninlined_format_args
```

Note that the MSRV must be at least 1.58 (which introduced inline format args)

BurntSushi commented:
> Currently the lint is in `pedantic`, but there are plans/hopes to move it to `style`.

Where can I participate in this discussion of plans?

workingjubilee commented on examples/raw.rs:
```diff
@@ -21,7 +21,7 @@ fn print() {
     backtrace::trace(|frame| {
         let ip = frame.ip();
-        print!("frame #{:<2} - {:#02$x}", cnt, ip as usize, HEX_WIDTH);
+        print!("frame #{cnt:<2} - {:#0HEX_WIDTH$x}", ip as usize);
```
This introduces active confusion into reading this print statement because these are now "out of order", i.e. I am no longer reading all the format arguments strictly "left to right".

workingjubilee commented on examples/raw.rs:
```diff
@@ -33,13 +33,13 @@ fn print() {
             if let Some(file) = symbol.filename() {
                 if let Some(l) = symbol.lineno() {
-                    print!("\n{:13}{:4$}@ {}:{}", "", "", file.display(), l, HEX_WIDTH);
+                    print!("\n{:13}{:HEX_WIDTH$}@ {}:{l}", "", "", file.display());
```
Same issue re: added confusion, ordering, and left-to-right reading.

workingjubilee commented on src/print.rs:
```diff
@@ -288,7 +288,7 @@ impl BacktraceFrameFmt<'_, '_, '_> {
         if self.symbol_index == 0 {
             self.fmt.fmt.write_str("{{{bt:")?;
-            write!(self.fmt.fmt, "{}:{:?}", self.fmt.frame_index, frame_ip)?;
+            write!(self.fmt.fmt, "{}:{frame_ip:?}", self.fmt.frame_index)?;
```
Same issue re: added confusion, ordering, and left-to-right reading.

workingjubilee commented on src/print.rs:
```diff
@@ -266,18 +266,18 @@ impl BacktraceFrameFmt<'_, '_, '_> {
         // Filename/line are printed on lines under the symbol name, so print
         // some appropriate whitespace to sort of right-align ourselves.
         if let PrintFmt::Full = self.fmt.format {
-            write!(self.fmt.fmt, "{:1$}", "", HEX_WIDTH)?;
+            write!(self.fmt.fmt, "{:HEX_WIDTH$}", "")?;
         }
         write!(self.fmt.fmt, "             at ")?;
```
If this part is going to be diffed, surely there's a better to write it, in general? The part you diffed is literally formatting the empty string in, and all this is to generate what is mostly whitespace...

workingjubilee reviewed (changes requested):
Thank you for your efforts. My apologies for it being so long to get around to this PR: this crate is fairly direly under-maintained. For that reason, currently this crate does not run clippy in CI, nor do I believe there is an intention to, as adding additional CI checks on getting code in is not what we want to be doing unless those are strictly necessary for functionality.

Normally I am inclined to approve a modest beautification PR like this, but I do not believe all of these changes are equally beautifying. Some make the code harder to read, by changing things so they are no longer read in a left-to-right manner, or in general complicating the code's overall style so that some of the results are surprising. If you wish to make additional changes to remove the diffs I have noted, or to improve the formatting code further until the concerns I have are eliminated, then I believe we could accept this. Otherwise, I will probably close this when I next make a pass over the PRs for this crate.

nyurik commented:
thx for great feedback, agree, will fix!

nyurik commented:
@workingjubilee I addressed your feedback, and added a few more similar inlinings.

To keep things cleaner, I think #539 should be merged first, possibly followed by #538

nyurik commented:
@workingjubilee i rebased this PR to make sure it is up to date. Please review & merge unless there are some new issues. Thx!
